src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -176,27 +178,34 @@ uct_ib_mlx5_devx_reg_ksm_data_contig(uct_ib_mlx5_md_t *md,
                                      void *address, int atomic,
                                      const char *reason,
                                      struct mlx5dv_devx_obj **mr_p,
-                                     uint32_t *mkey)
+                                     uint32_t *mkey,
+                                     uint8_t is_memic)
 {
     size_t mr_length = mr->super.ib->length;
     uintptr_t ksm_address;
     size_t ksm_length;
     int list_size;
 
-    /* FW requires indirect atomic MR address and length to be aligned
-     * to max supported atomic argument size */
-    ksm_address = ucs_align_down_pow2((uintptr_t)address,
-                                      UCT_IB_MD_MAX_MR_SIZE);
-    ksm_length  = mr_length + (uintptr_t)address - ksm_address;
-    ksm_length  = ucs_align_up(ksm_length, md->super.dev.atomic_align);
+    if (is_memic) {
+        ksm_address = (uintptr_t)address;
+        ksm_length  = mr_length;
```

Comment:
why needed? I'd expect memic address/length to be aligned anyway
besides, memic can also be used for RDMA read/write, not just atomics

Reply:
the address should be 0 for memic, maybe we can align the iova,
about the atomic alignment - it's not part of my change, this line is the same as in master

Reply:
Can we keep code as is and not have special handling for memic?